framework/src/base/Moose.C

```diff
@@ -328,6 +330,7 @@ addActionTypes(Syntax & syntax)
                            "(init_problem)"
                            "(add_control)"
                            "(check_output)"
+                           "(auto_checkpoint_action)"
```

Comment:
I didn't pay close-enough attention to this before. This action needs to go up on line 314, right after "add_output". The reason is that we assume that all outputs are in the system after that point. Any other Action that runs later is allowed to assume that no other Outputs will be added.